build-tools/packages/build-tools/src/fluidBuild/fluidUtils.ts

```diff
@@ -140,29 +144,37 @@ const configExplorer = cosmiconfigSync(configName, {
 	packageProp: [configName],
 });
 
+/**
+ * Set to true when the default config is returned by getFluidBuildConfig so that repeated calls to the function don't
+ * result in repeated searches for config.
+ */
+let defaultConfigLoaded = false;
+
 /**
  * Get an IFluidBuildConfig from the fluidBuild property in a package.json file, or from fluidBuild.config.[c]js.
  *
- * @param rootDir - The path to the root package.json to load.
- * @param noCache - If true, the config cache will be cleared and the config will be reloaded.
- * @returns The fluidBuild section of the package.json, or undefined if not found
+ * @param searchDir - The path to search for the config. The search will look up the folder hierarchy for a config in
+ * either a standalone file or package.json
+ * @returns The the loaded fluidBuild config, or the default config if one is not found.
  */
 export function getFluidBuildConfig(
-	rootDir: string,
-	noCache = false,
+	searchDir: string,
 	log = defaultLogger,
 ): IFluidBuildConfig {
-	if (noCache === true) {
-		configExplorer.clearCaches();
+	if (defaultConfigLoaded) {
+		return DEFAULT_FLUIDBUILD_CONFIG;
```

Comment:
So, you can only have one ... iff the first call resulted in default.
Otherwise, you can have multiple until the point one call results in default. After that all calls result in the default.

Hmmm.

Reply:
This seems correct. If a config is found, it will be returned, and repeated calls will return the same config - cosmicconfig has its own internal caching. If a config is not found, the default will be returned, and repeated calls will return the default config.

Reply:
The default check comes before the configExplorer search call. Once a config is found it will be used if there is a later search with different path?

Reply:
I see what you mean. Fortunately, I don't think it's a practical problem in the near term because fluid-build doesn't expect to get a different config per search path. Or, more precisely, fluid-build expects there to be a single config that is loaded for any path under its root, which, when there's no config, is the current working directory.

That said, an alternative pattern that I considered is returning undefined and making callers use the default config explicitly if that's the behavior they want. More or less what cosmicconfig itself does. I discarded it because I didn't like spreading the default logic around. But arguably the caller _should_ decide how to handle a missing config.

Maybe cosmicconfig caches searched paths that don't yield a config? If so, then maybe we could call explorer.search always, and only return the default when search doesn't find anything.